test/DaedalusCompiler.Tests/ParsingSourceToAbstractAssemblyTests.cs

```diff
@@ -5828,5 +5889,181 @@ class Person {
         }
+
+        [Fact]
+        public void TestSingleExpressionReturnHack()
+        {
+            _code = @"
+                func void testFunc() {
+                var int x;
+                    x = 5;
+                    x;
+                    return;
+                };
+           ";
+
+            _instructions = GetExecBlockInstructions("testFunc");
+            _expectedInstructions = new List<AssemblyElement>
+            {
+                // x = 5;
+                new PushInt(5),
+                new PushVar(Ref("testFunc.x")),
+                new Assign(),
+
+                // x;
+                new PushVar(Ref("testFunc.x")),
+
+                // return;
+                new Ret(),
+
+                new Ret()
+            };
+            AssertInstructionsMatch();
+
+            _expectedSymbols = new List<DatSymbol>
+            {
+                Ref("testFunc"),
+                Ref("testFunc.x"),
+            };
+            AssertSymbolsMatch();
+        }
+        
+        
+        [Fact]
+        public void TestInlineInstanceDeclarations()
+        {
+            _code = @"
+                class C_NPC { var int data [200]; };
+                instance Mage, Archer, Barbarian(C_NPC);
+
+                prototype Orc(C_NPC) {};
+                instance Warrior, Elite, Shaman(Orc);
+           ";
+
+            
+            _instructions = GetExecBlockInstructions("Mage");
+            _expectedInstructions = new List<AssemblyElement>
+            {
+                new Ret(),
+            };
+            AssertInstructionsMatch();
+            
+            
+            _instructions = GetExecBlockInstructions("Archer");
+            _expectedInstructions = new List<AssemblyElement>
+            {
+                new Ret(),
+            };
+            AssertInstructionsMatch();
+            
+            
+            _instructions = GetExecBlockInstructions("Barbarian");
+            _expectedInstructions = new List<AssemblyElement>
+            {
+                new Ret(),
+            };
+            AssertInstructionsMatch();
+            
+            
+            _instructions = GetExecBlockInstructions("Orc");
+            _expectedInstructions = new List<AssemblyElement>
+            {
+                new Ret(),
+            };
+            AssertInstructionsMatch();
+            
+            
+            _instructions = GetExecBlockInstructions("Warrior");
+            _expectedInstructions = new List<AssemblyElement>
+            {
+                new Call(Ref("Orc")),
+            };
+            AssertInstructionsMatch();
+            
+            
+            _instructions = GetExecBlockInstructions("Elite");
+            _expectedInstructions = new List<AssemblyElement>
+            {
+                new Call(Ref("Orc")),
+            };
+            AssertInstructionsMatch();
+            
+            
+            _instructions = GetExecBlockInstructions("Shaman");
```

Comment:
Why `Shaman` have a different body than  Elite and Warrior?

Reply:
Because original compiler sucks when you have `instance X, Y, Z(PROTOTYPE)` and generates 4 instructions from that: `001: Call(PROTOTYPE), 002: Call(PROTOTYPE), 003: Call(PROTOTYPE), 004: Ret()`, and X's first address is 001, Y's is 002, and Z's iz 003. Since there is no Ret instruction at the end of instructions for X and Y, when X will be created it will actually run Call(PROTOTYPE) 3 times. For now I made it like in the original compiler to have compare tests pass but in the I think we can change this behavior or not support syntax where you define multiple instances in one line.